rados/snapshot.go

```diff
@@ -0,0 +1,72 @@
+package rados
+
+// #cgo LDFLAGS: -lrados
+// #include <stdlib.h>
+// #include <rados/librados.h>
+import "C"
+
+import "unsafe"
+
+// Snapshot represents a snapshot on the particular ioctx.
+type Snapshot struct {
```

Comment:
Are you mimicking the pattern from rdb? Because I feel that's not a good pattern to emulate.
Things that are based purely on a string should just be functions against the ioctx and take string arguments. I feel pretty strongly about this. The api in rbd causes all sorts of weird coding patterns because of this.